b2share/modules/schemas/api.py

```diff
@@ -530,36 +530,36 @@ def create_version(cls, community_id, community_schema,
         from .models import CommunitySchemaVersion
 
         validate_json_schema(community_schema)
-        try:
-            with db.session.begin_nested():
-                try:
-                    last_schema = CommunitySchemaVersion.query.filter(
-                        CommunitySchemaVersion.community == community_id
-                    ).order_by(
-                        CommunitySchemaVersion.version.desc()
-                    ).limit(1).one()
-
-                    new_version = last_schema.version + 1
-                    if root_schema_version is None:
-                        root_schema_version = last_schema.root_schema
-                except NoResultFound:
-                    if root_schema_version is None:
-                        # there is no schema yet, the community is new.
-                        # Use the last RootSchema.
-                        root_schema_version = RootSchema.query.order_by(
-                            RootSchema.released.desc()).limit(1).one().id
-                    new_version = 0
-
-                model = CommunitySchemaVersion(
-                    community=community_id,
-                    root_schema=root_schema_version,
-                    community_schema=json.dumps(community_schema,
-                                                # avoid default whitespaces
-                                                separators=(',', ':')),
-                    version=new_version)
-                db.session.add(model)
-        except NoResultFound as e:
-            raise CommunitySchemaDoesNotExistError(id) from e
+        with db.session.begin_nested():
+            try:
+                last_schema = CommunitySchemaVersion.query.filter(
+                    CommunitySchemaVersion.community == community_id
+                ).order_by(
+                    CommunitySchemaVersion.version.desc()
+                ).limit(1).one()
+
+                new_version = last_schema.version + 1
+                if root_schema_version is None:
+                    root_schema_version = last_schema.root_schema
+            except NoResultFound:
+                if root_schema_version is None:
+                    # there is no schema yet, the community is new.
+                    # Use the last RootSchema.
+                    try:
+                        root_schema_version = RootSchemaVersion.query.order_by(
```

Comment:
This block is also changed in your recent PR, I think, so it will get another rewrite.